packages/manager/modules/vps/src/dashboard/tile/configuration/service.js

```diff
@@ -96,8 +96,15 @@ export default class ConfigurationTileService {
       };
     }
 
+    // Exclude all products which aren't listed in available upgrades
+    const availableCatalogProducts = this.catalog.products.filter((product) =>
+      this.availableUpgrades
+        .map((upgrade) => upgrade.planCode)
```

Comment:
Maybe we could map before filter function ? This way we will avoid to map for each element in `this.catalog.products`

Reply:
Sure :)